distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ServerCli.java

```diff
@@ -227,6 +240,17 @@ public void close() {
         if (server != null) {
             server.stop();
         }
+
+        // Wait for all post-exit processors to complete before shutting down
+        // Since close() is run inside the CliToolLauncher ShutdownHook, this will prevent the process from terminating (SIGTERM)
+        // until the work carried out by the post-exit processor(s) is finished
+        for (final var processor : serverProcessPostExitProcessors) {
```

Comment:
Initially I setup a specific runtime shutdown hook to do the waiting, but I realized this is not needed: the CliToolLauncher already calls close() in a shutdown hook, so we can take advantage of this and make the cli process wait for all post-processors to finish here, before exiting

Reply:
Could the serverless subclass for ServerCli override close, call super, then wait however it likes? No need for an additional interface.

Reply:
I think it's cleaner this way, but I see your point to avoid introducing another concept. Override close + call super will have the same effect.